### ISSUE TITLE
Every route passes the pattern check

### ISSUE THREAD

**Issue by aleksa2808** [GeniusesOfSymfony/PubSubRouterBundle#9]
I am trying out the channel route pattern requirement, and currently my routing.yml contains a route defined like this:

``` yaml
acme_topic:  
    channel: acme/{room}
    handler:
        callback: 'acme.topic'
    requirements:
        room:
            pattern: '\d+'
```

This works fine for routes defined like acme/5, but, unfortunately, it works even for routes defined as something arbitrary like acme/cat.

I have checked out the code behind the route matching and I found a few suspicious lines in `gos/pubsub-router-bundle/Matcher/Matcher.php' at line [128](https://github.com/GeniusesOfSymfony/PubSubRouterBundle/blob/master/Matcher/Matcher.php#L128):

``` php
$checkPattern = true;

//Wildcard requirements
if (isset($tokenRequirements['wildcard']) && true === $tokenRequirements['wildcard']) {
    if ($expectedToken->getExpression() === '*' || $expectedToken->getExpression() === 'all') {
        $this->validateToken($validTokens);
        $checkPattern = false;
    }
} else { // :127
    $this->validateToken($validTokens);
} // :129

//Pattern requirements
if (true === $checkPattern && isset($tokenRequirements['pattern'])) {
    $pattern = $tokenRequirements['pattern'];
    if (1 === preg_match("#^".$pattern."#i", $expectedToken->getExpression())) {
        $this->validateToken($validTokens);
    }
} else {
    $this->validateToken($validTokens);
}
```

The `else` branch after the wildcard token check will be accessed when the wildcard token is not set, or if it's set, but set to false. This means that if one of these is true (which should be, if you, like me, don't wish to add a wildcard route, but do want it to be pattern validated) the pattern validation code would be disregarded, since the `validateToken` function in the `else` branch already checks out the incorrect route.

tl;dr I believe that the code from line 127 to line 129 is incorrect.

Please respond if you think that I misunderstood the wildcard operator, or that this is the intended purpose of the Matcher class.

Thank you

**Comment by jjsaunier** [comment]
Yep you right, are you aware to provide a patch?

**Comment by aleksa2808** [comment]
Just made a pull request. This should fix the problem of pattern avoidance.
